3-0-spring-framework/3-1-1-dispatcher-servlet-initializer/src/test/java/com/bobocode/WelcomeWebAppTest.java

```diff
@@ -0,0 +1,173 @@
+    void webConfigIsMarkedAsConfiguration() {
+        Configuration configuration = WebConfig.class.getAnnotation(Configuration.class);
+
+        assertNotNull(configuration);
+    }
+
+    @Test
+    @Order(6)
+    @DisplayName("WebConfig enables @ComponentScan")
+    void webConfigEnablesComponentScan() {
+        ComponentScan componentScan = WebConfig.class.getAnnotation(ComponentScan.class);
+
+        assertNotNull(componentScan);
+    }
+
+    @Test
+    @Order(7)
+    @DisplayName("WebConfig @ComponentScan contains web packages")
+    void webConfigComponentScanPackages() {
+        ComponentScan componentScan = WebConfig.class.getAnnotation(ComponentScan.class);
+
+        assertThat(componentScan.basePackages()).contains("com.bobocode.web");
+    }
+
+    @Test
+    @Order(8)
+    @DisplayName("WebConfig is marked with @EnableWebMvc")
+    void webConfigEnablesWebMvc() {
+        EnableWebMvc enableWebMvc = WebConfig.class.getAnnotation(EnableWebMvc.class);
+
+        assertNotNull(enableWebMvc);
+    }
+
+    @Test
+    @Order(9)
+    @DisplayName("Initializer method is overridden with RootConfig class")
+    void initializerRootConfigClasses() {
+        WebAppInitializerWrapper webAppInitializerWrapper = new WebAppInitializerWrapper();
+
+        assertThat(webAppInitializerWrapper.getRootConfigClasses()).contains(RootConfig.class);
+    }
+
+    @Test
+    @Order(10)
+    @DisplayName("Initializer method is overridden with WebConfig class")
+    void initializerWebConfigClasses() {
+        WebAppInitializerWrapper webAppInitializerWrapper = new WebAppInitializerWrapper();
+
+        assertThat(webAppInitializerWrapper.getServletConfigClasses()).contains(WebConfig.class);
+    }
+
+    @Test
+    @Order(11)
+    @DisplayName("Proper Dispatcher Servlet mapping")
+    void dispatcherServletMapping() {
+        WebAppInitializerWrapper webAppInitializerWrapper = new WebAppInitializerWrapper();
+
+        assertThat(webAppInitializerWrapper.getServletMappings()).contains("/");
+    }
+
+    @Test
+    @Order(12)
+    @DisplayName("WelcomeController is marked as @Controller")
+    void welcomeControllerIsMarkedAsController() {
+        Controller controller = WelcomeController.class.getAnnotation(Controller.class);
+
+        assertNotNull(controller);
+    }
+
+    @Test
+    @Order(13)
+    @DisplayName("WelcomeController method is marked as Get method")
+    void welcomeControllerMethodIsMarkedAsGetMethod() throws NoSuchMethodException {
+        GetMapping getMapping = WelcomeController.class.getDeclaredMethod("welcome").getAnnotation(GetMapping.class);
+
+        assertNotNull(getMapping);
+    }
+
+    @Test
+    @Order(14)
+    @DisplayName("WelcomeController Get method is marked properly")
+    void welcomeControllerMethodMapping() throws NoSuchMethodException {
+        GetMapping getMapping = WelcomeController.class.getDeclaredMethod("welcome").getAnnotation(GetMapping.class);
```

Comment:
if you chain method calls, it's better to use separate lines for them or don't chain it at all.
```java
GetMapping getMapping = WelcomeController.class.getDeclaredMethod("welcome")
                .getAnnotation(GetMapping.class);
```
or
```java
Method welcomeMethod = WelcomeController.class.getDeclaredMethod("welcome");
GetMapping getMapping = weclomeMethod.getAnnotation(GetMapping.class);
```